buildSrc/src/main/java/org/elasticsearch/gradle/VersionProperties.java

```diff
@@ -27,11 +27,29 @@ public static String getLucene() {
     private static final Map<String, String> versions = new HashMap<String, String>();
     static {
         Properties props = getVersionProperties();
-        elasticsearch = Version.fromString(props.getProperty("elasticsearch"));
+        Version baseVersion = Version.fromString(props.getProperty("elasticsearch"));
+        if (baseVersion.isSnapshot()) {
+            throw new IllegalArgumentException("version.properties can't contain a snapshot version for elasticsearch. " +
+                "Use the `build.snapshot` property instead.");
+        }
+        if (baseVersion.getQualifier().isEmpty() == false) {
+            throw new IllegalArgumentException("version.properties can't contain a version qualifier for elasticsearch." +
+                "Use the `build.version_qualifier` property instead.");
+        }
+        elasticsearch = new Version(
+            baseVersion.getMajor(),
+            baseVersion.getMinor(),
+            baseVersion.getRevision(),
+            // TODO: Change default to "" after CI "warm-up" by adding -Dbuild.version_qualifier="" to ML jobs to produce
+            //       a snapshot.
```

Comment:
We now have the ability to set the ML C++ version qualifier when kicking off a build, so if it makes life easier for you I can create a one-off build of ML C++ 7.0.0-SNAPSHOT artifacts now so that the PR build can pass with a default qualifier of `""`.  Then ping me when you merge this PR and I'll change the value for the periodic ML C++ builds to be no qualifier too.